darwin/future/meta/objects/stage.py

```diff
@@ -66,7 +66,9 @@ def type(self) -> str:
     @property
     def edges(self) -> List[List[UUID]]:
         """Edge ID, source stage ID, target stage ID."""
+        assert self._element.edges is not None
         return [
             [edge.id, edge.source_stage_id, edge.target_stage_id]
             for edge in self._element.edges
+            if edge.source_stage_id and edge.target_stage_id
         ]
```

Comment:
@JBWilkie source_stage_id and target_stage_id are optional so they can be None. Which means this comprehension without the if check actually returns a List[List[UUID | None]]. Let me know if this change doesn't make sense though and you want the None's

Reply:
Can't see a use case for why the `None`s would be needed but you never know

Reply:
I actually changed this so that it returns the `WFEdgeCore` items instead of the IDs within them, not sure why I didn't do this originally

This means we don't need to worry about this if check